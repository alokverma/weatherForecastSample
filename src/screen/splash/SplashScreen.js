import React, { Component } from 'react'
import {View, Image, Text} from 'react-native';
import s from './styles';
import {THEMES} from '../../utils/constants'
const {colors} = THEMES.weatherDefaultTheme;

class SplashScreen extends Component{

    componentDidMount () {
      setTimeout (() => {
          this.props.navigation.navigate('Weather')
      }, 2000);
   }

    render() {
        return (
          <View style={s.container}>
            <Image
              style={{width: 300, height: 300}}
              source={require('../../assets/splash.png')}>
              </Image>
              <Text style = {s.textstyle} >WEATHER</Text>
            
          </View>
        )
      }
}


export default SplashScreen