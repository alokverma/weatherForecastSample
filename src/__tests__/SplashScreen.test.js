import React from 'react';
import {shallow} from 'enzyme';
import SplashScreen from '../screen/splash/SplashScreen';

describe('SplashScreen',()=>{
    describe('checking SnapShopt',()=>{
        it('should render splash screen  component',()=>{
            const component = shallow(<SplashScreen />)
            expect(component).toMatchSnapshot()
        });
    })
});