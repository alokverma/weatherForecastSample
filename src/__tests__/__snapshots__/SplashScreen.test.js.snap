// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SplashScreen checking SnapShopt should render splash screen  component 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../src/assets/splash.png",
      }
    }
    style={
      Object {
        "height": 300,
        "width": 300,
      }
    }
  />
  <Text
    style={
      Object {
        "color": "black",
        "fontSize": 40,
        "paddingBottom": 0,
        "textAlign": "center",
      }
    }
  >
    WEATHER
  </Text>
</View>
`;
